### PR TITLE
Misc: dependabot and golang version

### PR DESCRIPTION
**Pull request by lluiscampos**
* [pipeline] Minor beautification: image definition
* Introduce dependabot for pip and gitmodules
* Update golang to version 1.14.7

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -74,7 +74,7 @@ test:acceptance:
   script:
     - apk --update --no-cache add python3 gcc openssh make openssl-dev libffi-dev libc-dev python3-dev
     - cd tests
-    - pip3 install --upgrade -r requirements.txt
+    - pip3 install -r requirements.txt
```
Comment (oleorhagen) [inline .gitlab-ci.yml:77]: `Introduce dependabot and golang version` and... ?

**Review by oleorhagen — APPROVED** [review]
Nice !

**Review by lluiscampos — COMMENTED** [review]
Note to self: requires manual testing on Raspberry Pi Zero W

File: .gitlab-ci.yml (hunk 1 above)
Comment (lluiscampos) [reply .gitlab-ci.yml:77]: I consider this part of dependabot update; we use fixed requirements hence we don't need the upgrade flag.

Comment (lluiscampos) [reply .gitlab-ci.yml:77]: I'll add a comment in the commit message

**Pull request merged**